src/ProductConstructionService/ProductConstructionService.DependencyFlow/MergePolicyEvaluator.cs

```diff
@@ -33,12 +39,16 @@ public MergePolicyEvaluator(IEnumerable<IMergePolicyBuilder> mergePolicies, Oper
     public IImmutableDictionary<string, IMergePolicyBuilder> MergePolicyBuilders { get; }
     public ILogger<MergePolicyEvaluator> Logger { get; }
 
-    public async Task<MergePolicyEvaluationResults> EvaluateAsync(
+    public async Task<IEnumerable<MergePolicyEvaluationResult>> EvaluateAsync(
         PullRequestUpdateSummary pr,
         IRemote darc,
-        IReadOnlyList<MergePolicyDefinition> policyDefinitions)
+        IReadOnlyList<MergePolicyDefinition> policyDefinitions,
+        MergePolicyEvaluationResults? cachedResults,
+        string targetBranchSha)
     {
-        var results = new List<MergePolicyEvaluationResult>();
+        IDictionary<string, MergePolicyEvaluationResult> cachedResultsByPolicyName =
+            cachedResults?.Results.ToDictionary(r => r.MergePolicyName, r => r) ?? new Dictionary<string, MergePolicyEvaluationResult>();
```

Comment:
Would this work? I guess no reason to use interfaces when we have a concrete type already.

```suggestion
        Dictionary<string, MergePolicyEvaluationResult> cachedResultsByPolicyName =
            cachedResults?.Results.ToDictionary(r => r.MergePolicyName, r => r) ?? [];
```